### PR TITLE
[HttpKernel] Group deprecation notices

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | yes |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | - |
| License | MIT |
| Doc PR | - |

Let's remove duplicate deprecation messages.

capture du 2015-08-30 14 15 45

### CODE REVIEW

File: src/Symfony/Component/HttpKernel/DataCollector/LoggerDataCollector.php [hunk 1]
```diff
@@ -97,15 +97,43 @@ public function getName()
...
+                    }
+
+                    if (!$silenced && isset($errorContextById[$errorId]['scream'])) {
+                        unset($errorContextById[$errorId]['scream']);
+                    }
+
+                    continue;
+                }
+
+                $errorContextById[$errorId] = &$context;
```
Comment (sstok) [inline src/Symfony/Component/HttpKernel/DataCollector/LoggerDataCollector.php]: Is there a special reason for using a reference? :)

**Comment by sstok** [comment]
:+1: for this enhancement.

File: src/Symfony/Component/HttpKernel/DataCollector/LoggerDataCollector.php (hunk 1 above)
Comment (nicolas-grekas) [inline src/Symfony/Component/HttpKernel/DataCollector/LoggerDataCollector.php]: Yes: e.g. to increment the error count a few lines above

**Comment by iltar** [comment]
:+1: Nice, this will help a lot!

**Comment by nicolas-grekas** [comment]
Status: needs review
Ready, tests added

**Comment by fabpot** [comment]
What about merging this into 2.7 instead?

**Comment by nicolas-grekas** [comment]
Fine by me

**Comment by fabpot** [comment]
Thank you @nicolas-grekas.

File: src/Symfony/Component/HttpKernel/DataCollector/LoggerDataCollector.php [hunk 2]
```diff
@@ -97,15 +97,46 @@ public function getName()
...
-        foreach ($logs as $i => $log) {
+        $errorContextById = array();
+        $sanitizedLogs = array();
+
+        foreach ($logs as $log) {
             $context = $this->sanitizeContext($log['context']);
-            if (isset($context['type'], $context['level']) && !($context['type'] & $context['level'])) {
-                $context['scream'] = true;
+
+            if (isset($context['type'], $context['file'], $context['line'], $context['level'])) {
```
Comment (nicolas-grekas) [inline src/Symfony/Component/HttpKernel/DataCollector/LoggerDataCollector.php]: file and line are always set by the error handler so I'd say no need, do you think about something else?

Comment (stof) [inline src/Symfony/Component/HttpKernel/DataCollector/LoggerDataCollector.php]: ah, I forgot it was setting them automatically.

**Pull request closed without merging**